src/main.js

```diff
@@ -35,60 +37,77 @@ randomButton.addEventListener('click', function() {
 formButton.addEventListener('click', switchToFormView);
 viewSavedButton.addEventListener('click', switchToSavedView);
 homeButton.addEventListener('click', switchToHomeView);
+saveButton.addEventListener('click', saveVisibleCover)
 
-//event handlers
+// **** view switch functions ****
 function switchToFormView() {
-  homeView.classList.add('hidden'); //hide home view
-  savedView.classList.add('hidden'); //hide saved view
-  formView.classList.remove('hidden'); //reveal form view
-  homeButton.classList.remove('hidden'); //reveal home button
-  randomButton.classList.add('hidden'); //hide random cover button
-  saveButton.classList.add('hidden'); //hide saved cover button
+  homeView.classList.add("hidden"); //hide home view
+  savedView.classList.add("hidden"); //hide saved view
+  formView.classList.remove("hidden"); //reveal form view
+  homeButton.classList.remove("hidden"); //reveal home button
+  randomButton.classList.add("hidden"); //hide random cover button
+  saveButton.classList.add("hidden"); //hide saved cover button
 }
 
 function switchToSavedView() {
-  homeView.classList.add('hidden'); //hide home view
-  savedView.parentNode.classList.remove('hidden'); //reveal saved view
-  formView.classList.add('hidden'); //hide form view
-  homeButton.classList.remove('hidden'); //reveal home button
-  randomButton.classList.add('hidden'); //hide random cover button
-  saveButton.classList.add('hidden'); //hide saved cover button
+  homeView.classList.add("hidden"); //hide home view
+  savedView.classList.remove("hidden"); //reveal saved view
+  formView.classList.add("hidden"); //hide form view
+  homeButton.classList.remove("hidden"); //reveal home button
+  randomButton.classList.add("hidden"); //hide random cover button
+  saveButton.classList.add("hidden"); //hide saved cover button
+
+  var result = "";
```

Comment:
I think this part could be a helper function that's called on in switchToSavedView. I think separating them would better adhere to SRP prinicples